Fix flaky test: patmap scope

We have a flaky test in our test suite that is always making the CI fail. I have reproduced locally only once, but it's more prone to fail in the CI with the following log output:
```
Running source-dependencies / patMat-scope
557s
4028
[debug] Reading incremental options from map
561s
4029
[debug] USE_OPTIMIZED_SEALED value was read.
561s
4030
[debug] Full compilation, no sources in previous analysis.
561s
4031
[debug] All sources are invalidated.
561s
4032
[debug] Recompiling all 3 sources: invalidated sources (3) exceeded 50.0% of all sources
561s
4033
[info] Compiling 3 Scala sources to /tmp/sbt_a232f51a/patMat-scope/target/classes ...
561s
4034
[debug] Returning already retrieved and compiled bridge: /tmp/sbt_215f9b63/target-bridge-2.11.11.jar.
561s
4035
[debug] Returning already retrieved and compiled bridge: /tmp/sbt_215f9b63/target-bridge-2.11.11.jar.
561s
4036
[debug] [zinc] Running cached compiler 367e5d7f for Scala compiler version 2.11.11
561s
4037
[debug] [zinc] The Scala compiler is invoked with:
561s
4038
[debug] 	-bootclasspath
561s
4039
[debug] 	/usr/lib/jvm/jdk1.8.0_102/jre/lib/resources.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/rt.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/sunrsasign.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jsse.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jce.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/charsets.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jfr.jar:/usr/lib/jvm/jdk1.8.0_102/jre/classes:/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-library/scala-library-2.11.11.jar
561s
4040
[debug] 	-classpath
561s
4041
[debug] 	/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-compiler/scala-compiler-2.11.11.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-reflect/scala-reflect-2.11.11.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/bundles/org.scala-lang.modules/scala-xml_2.11/scala-xml_2.11-1.0.5.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/bundles/org.scala-lang.modules/scala-parser-combinators_2.11/scala-parser-combinators_2.11-1.0.4.jar:/tmp/sbt_a232f51a/patMat-scope/target/classes
561s
4042
[debug] Scala compilation took 0.280436274 s
561s
4043
[info] Done compiling.
561s
4044
[info] Compilation done: /tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/Sealed.scala, /tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/SealedUsedInPatMatScope.scala, /tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/SealedNameUsedInDefaultScope.scala
561s
4045
[debug] Reading incremental options from map
561s
4046
[debug] USE_OPTIMIZED_SEALED value was read.
561s
4047
[debug] 
561s
4048
[debug] Initial source changes: 
561s
4049
[debug] 	removed:Set()
561s
4050
[debug] 	added: Set()
561s
4051
[debug] 	modified: Set(/tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/Sealed.scala)
561s
4052
[debug] Invalidated products: Set()
561s
4053
[debug] External API changes: API Changes: Set()
561s
4054
[debug] Modified binary dependencies: Set()
561s
4055
[debug] Initial directly invalidated classes: Set(foo.Sealed, foo.SealedChild2, foo.SealedChild1, foo.SealedBase)
561s
4056
[debug] 
561s
4057
[debug] Sources indirectly invalidated by:
561s
4058
[debug] 	product: Set()
561s
4059
[debug] 	binary dep: Set()
561s
4060
[debug] 	external source: Set()
561s
4061
[debug] All initially invalidated classes: Set(foo.Sealed, foo.SealedChild2, foo.SealedChild1, foo.SealedBase)
561s
4062
[debug] All initially invalidated sources:Set(/tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/Sealed.scala)
561s
4063
[info] Compiling 1 Scala source to /tmp/sbt_a232f51a/patMat-scope/target/classes ...
561s
4064
[debug] Returning already retrieved and compiled bridge: /tmp/sbt_215f9b63/target-bridge-2.11.11.jar.
561s
4065
[debug] Returning already retrieved and compiled bridge: /tmp/sbt_215f9b63/target-bridge-2.11.11.jar.
561s
4066
[debug] [zinc] Running cached compiler 30955c70 for Scala compiler version 2.11.11
561s
4067
[debug] [zinc] The Scala compiler is invoked with:
561s
4068
[debug] 	-bootclasspath
561s
4069
[debug] 	/usr/lib/jvm/jdk1.8.0_102/jre/lib/resources.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/rt.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/sunrsasign.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jsse.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jce.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/charsets.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jfr.jar:/usr/lib/jvm/jdk1.8.0_102/jre/classes:/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-library/scala-library-2.11.11.jar
561s
4070
[debug] 	-classpath
561s
4071
[debug] 	/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-compiler/scala-compiler-2.11.11.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-reflect/scala-reflect-2.11.11.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/bundles/org.scala-lang.modules/scala-xml_2.11/scala-xml_2.11-1.0.5.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/bundles/org.scala-lang.modules/scala-parser-combinators_2.11/scala-parser-combinators_2.11-1.0.4.jar:/tmp/sbt_a232f51a/patMat-scope/target/classes
561s
4072
[debug] Scala compilation took 0.120076537 s
561s
4073
[info] Done compiling.
561s
4074
[debug] Invalidating (transitively) by inheritance from foo.SealedChild3...
561s
4075
[debug] Initial set of included nodes: Set(foo.SealedChild3)
561s
4076
[debug] Invalidated by transitive inheritance dependency: Set(foo.SealedChild3)
561s
4077
[debug] Change NamesChange(foo.SealedChild3,ModifiedNames(changes = UsedName(toString,[Default]), UsedName(getClass,[Default]), UsedName(SealedChild3,[Default]), UsedName(isInstanceOf,[Default]), UsedName(notifyAll,[Default]), UsedName(hashCode,[Default]), UsedName(asInstanceOf,[Default]), UsedName(equals,[Default]), UsedName($isInstanceOf,[Default]), UsedName(eq,[Default]), UsedName(synchronized,[Default]), UsedName(wait,[Default]), UsedName(ne,[Default]), UsedName($asInstanceOf,[Default]), UsedName(!=,[Default]), UsedName(finalize,[Default]), UsedName(clone,[Default]), UsedName(==,[Default]), UsedName(notify,[Default]), UsedName(##,[Default]))) invalidates 1 classes due to The foo.SealedChild3 has the following regular definitions changed:
561s
4078
[debug] 	UsedName(toString,[Default]), UsedName(getClass,[Default]), UsedName(SealedChild3,[Default]), UsedName(isInstanceOf,[Default]), UsedName(notifyAll,[Default]), UsedName(hashCode,[Default]), UsedName(asInstanceOf,[Default]), UsedName(equals,[Default]), UsedName($isInstanceOf,[Default]), UsedName(eq,[Default]), UsedName(synchronized,[Default]), UsedName(wait,[Default]), UsedName(ne,[Default]), UsedName($asInstanceOf,[Default]), UsedName(!=,[Default]), UsedName(finalize,[Default]), UsedName(clone,[Default]), UsedName(==,[Default]), UsedName(notify,[Default]), UsedName(##,[Default]).
561s
4079
[debug] 	> by transitive inheritance: Set(foo.SealedChild3)
561s
4080
[debug] 	> 
561s
4081
[debug] 	> 
561s
4082
[debug]         
561s
4083
[debug] Invalidating (transitively) by inheritance from foo.Sealed...
561s
4084
[debug] Initial set of included nodes: Set(foo.Sealed)
561s
4085
[debug] Invalidated by transitive inheritance dependency: Set(foo.Sealed)
561s
4086
[debug] None of the modified names appears in source file of foo.SealedNameUsedInDefaultScope. This dependency is not being considered for invalidation.
561s
4087
[debug] The following modified names cause invalidation of foo.SealedUsedInPatMatScope: Set(UsedName(Sealed,[Default, PatMatTarget]))
561s
4088
[debug] Change NamesChange(foo.Sealed,ModifiedNames(changes = UsedName(Sealed,[PatMatTarget]))) invalidates 2 classes due to The foo.Sealed has the following regular definitions changed:
561s
4089
[debug] 	UsedName(Sealed,[PatMatTarget]).
561s
4090
[debug] 	> by transitive inheritance: Set(foo.Sealed)
561s
4091
[debug] 	> 
561s
4092
[debug] 	> by member reference: Set(foo.SealedUsedInPatMatScope)
561s
4093
[debug]         
561s
4094
[debug] New invalidations:
561s
4095
[debug] 	Set(foo.SealedUsedInPatMatScope)
561s
4096
[debug] Initial set of included nodes: Set(foo.SealedUsedInPatMatScope)
561s
4097
[debug] Previously invalidated, but (transitively) depend on new invalidations:
561s
4098
[debug] 	Set()
561s
4099
[debug] All newly invalidated classes after taking into account (previously) recompiled classes:Set(foo.SealedUsedInPatMatScope)
561s
4100
[info] Compiling 1 Scala source to /tmp/sbt_a232f51a/patMat-scope/target/classes ...
561s
4101
[debug] Returning already retrieved and compiled bridge: /tmp/sbt_215f9b63/target-bridge-2.11.11.jar.
561s
4102
[debug] Returning already retrieved and compiled bridge: /tmp/sbt_215f9b63/target-bridge-2.11.11.jar.
561s
4103
[debug] [zinc] Running cached compiler 72a1632b for Scala compiler version 2.11.11
561s
4104
[debug] [zinc] The Scala compiler is invoked with:
561s
4105
[debug] 	-bootclasspath
561s
4106
[debug] 	/usr/lib/jvm/jdk1.8.0_102/jre/lib/resources.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/rt.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/sunrsasign.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jsse.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jce.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/charsets.jar:/usr/lib/jvm/jdk1.8.0_102/jre/lib/jfr.jar:/usr/lib/jvm/jdk1.8.0_102/jre/classes:/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-library/scala-library-2.11.11.jar
561s
4107
[debug] 	-classpath
561s
4108
[debug] 	/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-compiler/scala-compiler-2.11.11.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/jars/org.scala-lang/scala-reflect/scala-reflect-2.11.11.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/bundles/org.scala-lang.modules/scala-xml_2.11/scala-xml_2.11-1.0.5.jar:/drone/src/github.com/sbt/zinc/./target/lib_managed/bundles/org.scala-lang.modules/scala-parser-combinators_2.11/scala-parser-combinators_2.11-1.0.4.jar:/tmp/sbt_a232f51a/patMat-scope/target/classes
561s
4109
[debug] Scala compilation took 0.121357671 s
561s
4110
[info] Done compiling.
561s
4111
[debug] New invalidations:
561s
4112
[debug] 	Set()
561s
4113
[debug] Initial set of included nodes: Set()
561s
4114
[debug] Previously invalidated, but (transitively) depend on new invalidations:
561s
4115
[debug] 	Set()
561s
4116
[debug] All newly invalidated classes after taking into account (previously) recompiled classes:Set()
561s
4117
[info] Compilation done: /tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/Sealed.scala, /tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/SealedUsedInPatMatScope.scala, /tmp/sbt_a232f51a/patMat-scope/src/main/scala/foo/SealedNameUsedInDefaultScope.scala
561s
4118
[error] x source-dependencies / patMat-scope
561s
4119
sbt.internal.scripted.TestException: {line 16}  Command failed
561s
4120
	at sbt.internal.scripted.ScriptRunner.sbt$internal$scripted$ScriptRunner$$processStatement$1(ScriptRunner.scala:25)
561s
4121
	at sbt.internal.scripted.ScriptRunner$$anonfun$apply$2.apply(ScriptRunner.scala:40)
561s
4122
	at sbt.internal.scripted.ScriptRunner$$anonfun$apply$2.apply(ScriptRunner.scala:40)
561s
4123
	at scala.Function$$anonfun$tupled$1.apply(Function.scala:77)
561s
4124
	at scala.Function$$anonfun$tupled$1.apply(Function.scala:76)
561s
4125
	at scala.collection.immutable.List.foreach(List.scala:392)
561s
4126
	at sbt.internal.scripted.ScriptRunner.apply(ScriptRunner.scala:40)
561s
4127
	at sbt.internal.scripted.ScriptedTests.runTest$1(ScriptedTests.scala:106)
561s
4128
	at sbt.internal.scripted.ScriptedTests.sbt$internal$scripted$ScriptedTests$$scriptedTest(ScriptedTests.scala:115)
561s
4129
	at sbt.internal.scripted.ScriptedTests$$anonfun$scriptedTest$2$$anonfun$apply$2$$anonfun$apply$3$$anonfun$apply$4.apply(ScriptedTests.scala:71)
561s
4130
	at sbt.internal.scripted.ScriptedTests$$anonfun$scriptedTest$2$$anonfun$apply$2$$anonfun$apply$3$$anonfun$apply$4.apply(ScriptedTests.scala:65)
561s
4131
	at sbt.internal.io.Resources.sbt$internal$io$Resources$$readWrite$1(Resources.scala:57)
561s
4132
	at sbt.internal.io.Resources$$anonfun$readWriteResourceDirectory$1.apply(Resources.scala:59)
561s
4133
	at sbt.internal.io.Resources$$anonfun$readWriteResourceDirectory$1.apply(Resources.scala:59)
561s
4134
	at sbt.io.IO$.withTemporaryDirectory(IO.scala:332)
561s
4135
	at sbt.io.IO$.withTemporaryDirectory(IO.scala:340)
561s
4136
	at sbt.internal.io.Resources.readWriteResourceDirectory(Resources.scala:59)
561s
4137
	at sbt.internal.io.Resources.readWriteResourceDirectory(Resources.scala:47)
561s
4138
	at sbt.internal.scripted.ScriptedTests$$anonfun$scriptedTest$2$$anonfun$apply$2$$anonfun$apply$3.apply(ScriptedTests.scala:65)
561s
4139
	at sbt.internal.scripted.ScriptedTests$$anonfun$scriptedTest$2$$anonfun$apply$2$$anonfun$apply$3.apply(ScriptedTests.scala:63)
561s
4140
	at sbt.internal.scripted.ScriptedRunnerImpl$$anonfun$2.apply(ScriptedTests.scala:25)
561s
4141
	at sbt.internal.scripted.ScriptedRunnerImpl$$anonfun$2.apply(ScriptedTests.scala:25)
561s
4142
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
561s
4143
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
561s
4144
	at scala.collection.mutable.ArraySeq.foreach(ArraySeq.scala:74)
561s
4145
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
561s
4146
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
561s
4147
	at sbt.internal.scripted.ScriptedRunnerImpl$.runAll(ScriptedTests.scala:25)
561s
4148
	at sbt.internal.scripted.ScriptedRunnerImpl$.run(ScriptedTests.scala:22)
561s
4149
	at sbt.internal.inc.IncScriptedRunner$$anonfun$run$1.apply(IncScriptedRunner.scala:13)
561s
4150
	at sbt.internal.inc.IncScriptedRunner$$anonfun$run$1.apply(IncScriptedRunner.scala:10)
561s
4151
	at sbt.io.IO$.withTemporaryDirectory(IO.scala:332)
561s
4152
	at sbt.io.IO$.withTemporaryDirectory(IO.scala:340)
561s
4153
	at sbt.internal.inc.IncScriptedRunner.run(IncScriptedRunner.scala:10)
561s
4154
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
561s
4155
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
561s
4156
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
561s
4157
	at java.lang.reflect.Method.invoke(Method.java:498)
561s
4158
	at Scripted$.doScripted(Scripted.scala:82)
561s
4159
	at $c164b9d310a74142dd99$.$anonfun$scriptedTask$5(build.sbt:472)
561s
4160
	at $c164b9d310a74142dd99$.$anonfun$scriptedTask$5$adapted(build.sbt:459)
561s
4161
	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
561s
4162
	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:40)
561s
4163
	at sbt.std.Transform$$anon$4.work(System.scala:64)
561s
4164
	at sbt.Execute.$anonfun$submit$2(Execute.scala:257)
561s
4165
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
561s
4166
	at sbt.Execute.work(Execute.scala:266)
561s
4167
	at sbt.Execute.$anonfun$submit$1(Execute.scala:257)
561s
4168
	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:167)
561s
4169
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:32)
561s
4170
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
561s
4171
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
561s
4172
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
561s
4173
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
561s
4174
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
561s
4175
	at java.lang.Thread.run(Thread.java:745)
561s
4176
Caused by: java.lang.RuntimeException: Test script error: /tmp/sbt_a232f51a/patMat-scope/target/classes/foo/SealedUsedInPatMatScope.class is not newer than /tmp/sbt_a232f51a/patMat-scope/beforeFirstCompilation
561s
4177
	at scala.sys.package$.error(package.scala:27)
561s
4178
	at sbt.internal.scripted.FileCommands.scriptError(FileCommands.scala:45)
561s
4179
	at sbt.internal.scripted.FileCommands.newer(FileCommands.scala:70)
561s
4180
	at sbt.internal.scripted.FileCommands$$anonfun$commandMap$7.apply(FileCommands.scala:23)
561s
4181
	at sbt.internal.scripted.FileCommands$$anonfun$commandMap$7.apply(FileCommands.scala:23)
561s
4182
	at sbt.internal.scripted.FileCommands$CommandBuilder$$anonfun$twoArg$1.apply(FileCommands.scala:107)
561s
4183
	at sbt.internal.scripted.FileCommands$CommandBuilder$$anonfun$twoArg$1.apply(FileCommands.scala:106)
561s
4184
	at sbt.internal.scripted.FileCommands$$anonfun$1.apply(FileCommands.scala:40)
561s
4185
	at sbt.internal.scripted.FileCommands$$anonfun$1.apply(FileCommands.scala:40)
561s
4186
	at scala.Option.map(Option.scala:146)
561s
4187
	at sbt.internal.scripted.FileCommands.apply(FileCommands.scala:40)
561s
4188
	at sbt.internal.scripted.BasicStatementHandler$class.apply(StatementHandler.scala:18)
561s
4189
	at sbt.internal.scripted.FileCommands.apply(FileCommands.scala:13)
561s
4190
	at sbt.internal.scripted.FileCommands.apply(FileCommands.scala:13)
561s
4191
	at sbt.internal.inc.SleepingHandler.apply(ScriptedHandlers.scala:12)
561s
4192
	at sbt.internal.scripted.ScriptRunner.sbt$internal$scripted$ScriptRunner$$processStatement$1(ScriptRunner.scala:18)
561s
4193
	... 55 more
```
  
The fact that this test is failing is weird. It seems that for some reason the compiled products are not picked up, or the file system is misbehaving and not synchronizing to disk. The thing is, something's up and we need to fix it so that our CI passes always.

--- a/zinc/src/sbt-test/source-dependencies/patMat-scope/test
+++ b/zinc/src/sbt-test/source-dependencies/patMat-scope/test
@@ -5,6 +5,8 @@ $ exists target/classes/foo/SealedNameUsedInDefaultScope.class
 # Type of one child of Sealed changed
 $ copy-file changed/Sealed1.scala src/main/scala/foo/Sealed.scala
 $ touch beforeFirstCompilation
+# wait a second to be sure that modifaction date is different on all OSs
+$ sleep 1000
 
 > compile
 $ exists target/classes/foo/SealedUsedInPatMatScope.class
@@ -18,6 +20,8 @@ $ newer target/classes/foo/SealedUsedInPatMatScope.class beforeFirstCompilation
 # Change type of Sealed
 $ copy-file changed/Sealed2.scala src/main/scala/foo/Sealed.scala
 $ touch beforeSecondCompilation
+# wait a second to be sure that modifaction date is different on all OSs
+$ sleep 1000
 
 > compile
 $ exists target/classes/foo/SealedUsedInPatMatScope.class
@@ -26,4 +30,3 @@ $ exists target/classes/foo/SealedNameUsedInDefaultScope.class
 # Both PatMat and Default scopes should be recompiled
 $ newer target/classes/foo/SealedNameUsedInDefaultScope.class beforeSecondCompilation
 $ newer target/classes/foo/SealedUsedInPatMatScope.class beforeSecondCompilation
-

